### PR TITLE
Feature expr2c extension

### PR DESCRIPTION
**Pull request by mmuesly**
I tried to convert an exprt to compilable c code, but encountered some problems as struct type definitions and padding struct members are contained in the output.

This changes add a new class, that is capable to remove them from the conversion output.

Changing the behavior of expr2ct with parameters doesn't seem an easily feasible approach.

Further, "TRUE" and "FALSE" are not understood by the c compiler in the default case. I suggest to replace them with some accepted replacement values.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 659aa4495423b807d526475850aeb2aff5d39596).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80160965

File: src/ansi-c/expr2c.cpp [hunk 1]
```diff
@@ -1998,14 +1918,12 @@ std::string expr2ct::convert_constant_bool(bool boolean_value)
 {
   // C doesn't really have these
   if(boolean_value)
-    return "TRUE";
+    return "1";
```
Comment (mmuesly) [inline src/ansi-c/expr2c.cpp:1921]: Changing the boolean value was the original reason for this commit.

I am sorry for the reformatting around this change in the commit.

My intend was to change TRUE to 1 and FALSE to 0.

**Comment by peterschrammel** [comment]
I think @thk123 has a patch that subsumes these changes. @thk123, can you have a look into upstreaming expr2cleanc please?

**Comment by tautschnig** [comment]
Some of the changes here are more appropriate and/or overlapping changes with #2001. But if there is a much cleaner solution (expr2cleanc?) then this will be much appreciated! Assigning to @thk123.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 131bb8cf0eab320d27d17ffd81853db692ffa42a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80535721
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 1381399e31329a3c7bfc11ca95f484f177ad3890).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80797564
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Comment by mmuesly** [comment]
@thk123 Have you had a chance to take a look on this? I might provide some support in merging this with other PR into a joined version und your guidance, if that helps.

I really would appreciate, if the output of expr2ct becomes compilable soon.

**Comment by thk123** [comment]
@mmuesly sorry for the delay - done in #2704

**Comment by mmuesly** [comment]
Thanks, so I will close this PR in favor of #2704.

**Pull request closed without merging**